ubuntu22.04/precompiled/nvidia-driver

```diff
@@ -124,6 +143,10 @@ _load_driver() {
 
     DRIVER_VERSION=$(nvidia-smi -q | grep "Driver Version" | awk -F: '{print $2}' | xargs)
     if _assert_nvswitch_system; then
+        echo "Installing NVIDIA fabric manager and libnvidia NSCQ packages..."
+        apt-get install -y --no-install-recommends \
+            nvidia-fabricmanager-${DRIVER_BRANCH}=${DRIVER_VERSION}-1 \
+            libnvidia-nscq-${DRIVER_BRANCH}=${DRIVER_VERSION}-1
```

Comment:
Question -- should this be in `_install_driver` instead?

Reply:
I thought it's better to install it here since we check for n switch devices and we also grab the DRIVER_VERSION from nvidia-smi in this method

Reply:
> I understand moving the installation of fabricmanager / nscq to _install_driver() would result us in calling _assert_nvswitch_system() twice

We are actually only calling it once. Right now, the _assert_nvswitch_system() was only used to check whether to start nv-fabricmanager or not, now we are also installing the downloaded fabricmanager and nscq packages

> Isn't DRIVER_VERSION already set as an ENV in the Dockerfile? So why do we even need to resolve it on line 144? 

Correct,  I just thought the `nvidia-smi` output is more reliable as it's fetching the actual version of the driver installed via the precompiled package. Moreover, we already do this in the ubuntu20.04 implementation. See [here](https://github.com/NVIDIA/gpu-driver-container/blob/84bff94ceb321bb9e75ec49bd95f309ece1caf74/ubuntu20.04/precompiled/nvidia-driver#L126)

Reply:
Got it. I am fine with this implementation then.

Reply:
Thanks!